ugs-core/src/com/willwinder/universalgcodesender/uielements/helpers/FontManager.java

```diff
@@ -0,0 +1,148 @@
+/*
+    Copyright 2016-2017 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.universalgcodesender.uielements.helpers;
+
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import java.awt.Font;
+import java.awt.GraphicsEnvironment;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FontManager {
```

Comment:
This is pretty specific to the DRO panel right now, were you thinking it would evolve into a general purpose thing? Could it be renamed to MachineStatusFontManager for now?

Reply:
I expect to use this in the jog panel as well. I first started with this as a Singleton - no need to load the fonts more than once. Then I started adding listeners and such. I don't have a problem calling it MachineStatusFontManager and then just copying it for jog panel use. I'll see if I can figure out something better.

Reply:
Marking as done for tracking my tasks.